### PR TITLE
compact-all-views: wait until compaction ends

### PR DESCRIPTION
**Pull request by clochix**

### CODE REVIEW

File: lib/database.coffee [hunk 1]
```diff
@@ -27,20 +27,32 @@ request = require("request-json-light")
...
+        view: "view_compaction"
     setTimeout ->
         client.get '_active_tasks', (err, res, body) ->
-            exist = false
-            for task in body
-                if task.type is "database_compaction"
-                    exist = true
-            if (not exist) and found
-                callback true
+            console.log body
```
Comment (nono) [inline lib/database.coffee]: @clochix `console.log` still here

File: lib/database.coffee [hunk 2]
```diff
@@ -27,20 +27,32 @@ request = require("request-json-light")
...
-    couchClient.setBasicAuth username, password
+    # Only set auth if database has a password
+    if username or password
+        couchClient.setBasicAuth username, password
 
 # Wait end of compaction
-waitCompactComplete = (client, found, callback) ->
+# First must be true on first call, false on subsequents, otherwise, if
+# compaction takes less than 500ms, this function never returns
+waitCompactComplete = (client, found, type, callback, first = true) ->
```
Comment (nono) [inline lib/database.coffee]: @clochix the convention is to have the callback as last parameter

File: lib/database.coffee [hunk 3]
```diff
@@ -92,19 +106,21 @@ module.exports.compact = (database, callback)->
         if err or not body.ok
             callback makeError(err, body)
         else
-            waitCompactComplete couchClient, false, (success) ->
+            waitCompactComplete couchClient, false, "base", callback
```
Comment (nono) [inline lib/database.coffee]: @clochix I'm not sure here. Normally, a callback like this `callback` expects an err in first paramater. But it was `success` before your change. Are you sure that it won't cause trouble?

**Comment by clochix** [comment]
I removed the log and updated parameters order.
The `success` parameter was never used, that's why I removed it.

**Comment by clochix** [comment]
BTW, the failing tests have nothing to do with code I updated. Maybe a problem with the DS ?

**Comment by clochix** [comment]
Ping @poupotte for review of 67329db

**Comment by clochix** [comment]
@poupotte I now check if all stack applications are up to date before removing the notification (well, I hope I do).

**Comment by clochix** [comment]
@poupotte also, I fear this one may conflict with #112

**Comment by clochix** [comment]
Of course, thanks for noticing @poupotte, I just fixed this ☺

**Pull request merged**